lib/matplotlib/text.py

```diff
@@ -1380,6 +1386,30 @@ def set_fontname(self, fontname):
         """
         self.set_fontfamily(fontname)
 
+    def ha_for_angle(self, angle):
```

Comment:
`angle` here and below should be restricted to [0, 360) deg. Docstring should also call out that `angle` is measured in degrees and not radians

Reply:
I don't think this is being covered yet - the angle input here and the va function should be wrapped to [0, 360). In other words: `angle = angle % 360`

Reply:
Sorry if I didn't mention this earlier! Matplotlib already applies the modulo operation to the input angle before it reaches `_ha_for_angle` and `_va_for_angle`. It applies the modulo operation here:

```python 
 def set_rotation(self, s):
        """
        Set the rotation of the text.

        Parameters
        ----------
        s : float or {'vertical', 'horizontal'}
            The rotation angle in degrees in mathematically positive direction
            (counterclockwise). 'horizontal' equals 0, 'vertical' equals 90.
        """
        if isinstance(s, Real):
            self._rotation = float(s) % 360
        elif cbook._str_equal(s, 'horizontal') or s is None:
            self._rotation = 0.
        elif cbook._str_equal(s, 'vertical'):
            self._rotation = 90.
        else:
            raise ValueError("rotation must be 'vertical', 'horizontal' or "
                             f"a number, not {s}")
        self.stale = True
```

#### Example
Temporarily adding print statements:
```python 
def _ha_for_angle(self, angle, is_tick_top_enabled):
        """
        Determines horizontal alignment ('ha') based on the angle of rotation
        in degrees. Adjusts for is_tick_top_enabled.
        """
        print('angle:', angle)
        if (angle < 5 or 85 <= angle < 105 or 355 <= angle < 360 or
                170 <= angle < 190 or 265 <= angle < 275):
            return 'center'
        elif 5 <= angle < 85 or 190 <= angle < 265:
            return 'left' if is_tick_top_enabled else 'right'
        return 'right' if is_tick_top_enabled else 'left'

    def _va_for_angle(self, angle, is_tick_right_enabled):
        """
        Determines vertical alignment ('va') based on the angle of rotation
        in degrees. Adjusts for is_tick_right_enabled.
        """
        print('angle:', angle)
        if (angle < 5 or 355 <= angle < 360 or 170 <= angle < 190
                or 85 <= angle < 105 or 265 <= angle < 275):
            return 'center'
        elif 190 <= angle < 265 or 5 <= angle < 85:
            return 'baseline' if is_tick_right_enabled else 'top'
        return 'top' if is_tick_right_enabled else 'baseline'
```
Exercising the code path:
```python 
import matplotlib.pyplot as plt
import numpy as np

angles, n_cols = np.linspace(-45, 405, 2), 2
fig, axs = plt.subplots(nrows=1, ncols=n_cols, figsize=(4, 2.5))
axs = axs.flatten()

for i, angle in enumerate(angles):
    ax = axs[i]
    ax.set_title(f'Rotation: {angle:.0f}°', fontsize=10)

    ax.set_xticks([0.5])
    ax.set_xticklabels(['L'])
    ax.xaxis.set_tick_params(rotation=angle, rotation_mode='xtick')

    ax.set_yticks([0.5])
    ax.set_yticklabels(['L'])
    ax.yaxis.set_tick_params(rotation=angle, rotation_mode='ytick')

plt.savefig('rotated_ticks_figure.png')
plt.show()
```
![rotated_ticks_figure](https://github.com/user-attachments/assets/294913ab-1bc3-482d-8e50-45b3413a7c5b)

Terminal:
```
angle: 315.0
angle: 45.0
```
I restricted the angle to [0,360) from [0,360] based on your suggestion. If you want the modulo operations to be added to `_va_for_angle` and `_ha_for_angle` as an additional failsafe, let me know. Thanks!